recipes/parlayhash/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(test_package LANGUAGES CXX)
+
+find_package(parlayhash REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE parlay)
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
```

```suggestion
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
```

Because `std::result_of` has been removed since C++20, 
we have to specify using C++17.

https://en.cppreference.com/w/cpp/types/result_of

---

I think this should be part of the validate function - I have updated it accordingly, but I realized that this would affect every configuration, not just apple-clang, so I wonder why I can't compile in `apple-clang` for `compiler.cppstd=20`, but the same passes in Linux